robolectric-shadows/shadows-core/src/main/resources/org/robolectric/fakes/RenderServiceFactory.java.vm

```diff
@@ -0,0 +1,263 @@
+        width = size1 < size2 ? size2 : size1;
+        height = size1 < size2 ? size1 : size2;
+        break;
+      case PORTRAIT:
+        width = size1 < size2 ? size1 : size2;
+        height = size1 < size2 ? size2 : size1;
+        break;
+      case SQUARE:
+        width = height = size1;
+        break;
+    }
+    int wdp = (width * Density.DEFAULT_DENSITY) / density.getDpiValue();
+    int hdp = (height * Density.DEFAULT_DENSITY) / density.getDpiValue();
+    config.addQualifier(new SmallestScreenWidthQualifier(wdp < hdp ? wdp : hdp));
+    config.addQualifier(new ScreenWidthQualifier(wdp));
+    config.addQualifier(new ScreenHeightQualifier(hdp));
+    config.addQualifier(new ScreenSizeQualifier(screenSize));
+    config.addQualifier(new ScreenRatioQualifier(screenRatio));
+    config.addQualifier(new ScreenOrientationQualifier(orientation));
+    config.addQualifier(new DensityQualifier(density));
+    config.addQualifier(new TouchScreenQualifier(touchScreen));
+    config.addQualifier(new KeyboardStateQualifier(keyboardState));
+    config.addQualifier(new TextInputMethodQualifier(keyboard));
+    config.addQualifier(new NavigationStateQualifier(navigationState));
+    config.addQualifier(new NavigationMethodQualifier(navigation));
+    config.addQualifier(
+        width > height
+            ? new ScreenDimensionQualifier(width, height)
+            : new ScreenDimensionQualifier(height, width));
+    config.addQualifier(new VersionQualifier(apiLevel));
+    config.updateScreenWidthAndHeight();
+    return config;
+  }
+
+  /**
+   * Returns a {@link ResourceResolver} for a given config and project resource.
+   *
+   * @param config
+   * @param projectResources
+   * @param themeName
+   * @param isProjectTheme
+   * @return
+   */
+  public ResourceResolver createResourceResolver(FolderConfiguration config,
+      ResourceRepository projectResources, String themeName, boolean isProjectTheme) {
+    Map<ResourceType, Map<String, ResourceValue>> configedProjectRes =
+        projectResources.getConfiguredResources(config);
+    Map<ResourceType, Map<String, ResourceValue>> configedFrameworkRes =
+        mResources.getConfiguredResources(config);
+    return ResourceResolver.create(
+        configedProjectRes, configedFrameworkRes, themeName, isProjectTheme);
+  }
+
+  /**
+   * Creates a RenderService
+   *
+   * @param resources
+   * @param config
+   * @param projectCallback
+   * @return
+   */
+  public RenderService createService(
+      ResourceResolver resources, FolderConfiguration config, IProjectCallback projectCallback) {
+    RenderService renderService = new RenderService(mLibrary, resources, config, projectCallback);
+    return renderService;
+  }
+
+  /**
+   * Creates a RenderService. This is less efficient than
+   * {@link #createService(ResourceResolver, FolderConfiguration, IProjectCallback)} since the
+   * {@link ResourceResolver} object is not cached by the caller.
+   *
+   * @param projectResources
```

Comment:
They're still here. Can you go through and either document params where appropriate or just remove them from the doc, in most cases I think its self explanatory.